mesh/mesh.cpp

```diff
@@ -10751,16 +10751,19 @@ GeometricFactors::GeometricFactors(const Mesh *mesh, const IntegrationRule &ir,
    if (flags & GeometricFactors::COORDINATES)
    {
       X.SetSize(vdim*NQ*NE);
+      X.UseTemporary(use_temp_mem);
```

Comment:
Minor comment, I was thinking this could be ``UseDevicePool`` or something of the sort since the following vector would also go away after it goes out of scope. 
```
{
  Vector X; 
}
````

Reply:
Actually, use device pool may not make sense since you could have a permanent device pool, so temporary may make more sense.